src/pages/auth/sign_up/SignUp.jsx

```diff
@@ -0,0 +1,129 @@
+import { useState } from "react";
+import {  useNavigate } from "react-router-dom";
+import { ButtonPrimary, InputField, PasswordInput } from "../../../components";
+import { useToast, useAuth } from "../../../context";
+import { signUp } from "../../../services";
+
+export const SignUp = () => {
+  const navigate = useNavigate();
+  const { setIsUserLoggedIn, userDataDispatch } = useAuth();
+  const { showToast } = useToast();
+  const [inputValues, setInputValues] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
+  const { firstName, lastName, email, password } = inputValues;
+  const [agreeToTerms, setAgreeToTerms] = useState(false);
+
+  /* email and password validation */
+  var passwordPattern =
+    /^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{9,16}$/;
+  const validPassword = passwordPattern.test(password);
+
+  const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+  const validEmail = emailPattern.test(email);
+
+  const canSubmit = () => {
+    return firstName !== "" && lastName !== "" && validEmail && validPassword;
+  };
+  /* if agree to terms and can submit then set disabled false */
+  const isDisabled = () => !(agreeToTerms && canSubmit());
+  const disabled = isDisabled();
+
```

Comment:
Appreciation: Nice use of regex in validation

Reply:
Thank you